src/test/java/uk/gov/ons/ctp/response/notify/client/ConfigurationAwareNotificationClientTests.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.ons.ctp.response.notify.client;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import uk.gov.ons.ctp.response.notify.client.ConfigurationAwareNotificationClient;
+import uk.gov.ons.ctp.response.notify.config.NotifyConfiguration;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientApi;
+import uk.gov.service.notify.NotificationClientException;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ConfigurationAwareNotificationClientTests {
+    private static final String DUMMY_API_KEY = "dummykey-ffffffff-ffff-ffff-ffff-ffffffffffff-ffffffff-ffff-ffff-ffff-ffffffffffff";
+    private static final String DUMMY_TEMPLATE_ID = "ffffffff-ffff-ffff-ffff-ffffffffffff";
+
+    @Mock
+    private NotificationClient realNotificationClient;
+
+    @Mock
+    private NotifyConfiguration notifyConfiguration;
+
+    private NotificationClientApi testNotificationClient;
+
+    @Before
+    public void setup(){
+        when(notifyConfiguration.getApiKey()).thenReturn(DUMMY_API_KEY);
+        when(notifyConfiguration.getCensusUacSmsTemplateId()).thenReturn(DUMMY_TEMPLATE_ID);
+        when(notifyConfiguration.getOnsSurveysRasEmailReminderTemplateId()).thenReturn(DUMMY_TEMPLATE_ID);
+
+        this.testNotificationClient = new ConfigurationAwareNotificationClient(this.notifyConfiguration, this.realNotificationClient);
+    }
+
+    @Test
+    public void testAllApiEnabled() throws NotificationClientException {
```

Comment:
Would be cleaner to separate these out into different tests? It might make it easier to identify if a test fails. Same comment applies to `testAllApiDisabled`

Reply:
Done!